pkg/controllers/user/clusterauthtoken/token.go

```diff
@@ -111,25 +100,26 @@ func (h *TokenHandler) Remove(token *managementv3.Token) (runtime.Object, error)
 	if err != nil {
 		return nil, err
 	}
+	err = h.clusterAuthToken.Delete(token.Name, &metav1.DeleteOptions{})
+	if errors.IsNotFound(err) {
```

Comment:
This should only return if the errors isn't `IsNotFound`, otherwise you want to attempt the clusterUserAttribute cleanup.